File: rps/solutions/rps_random.py
import random

# Solutions should be deterministic, so we need to set our random seed
random.seed(1337)

def play_move() -> str:
    return random.choice('RPS')

def read_bot_move(bot_move: str) -> None:
    # This strategy doesn't care about opponent moves, so we do nothing here
    return

def main():
    T = int(input())
    for _ in range(T):
        print(play_move())
        read_bot_move(input())

if __name__ == '__main__':
    main()

File: rps/solutions/RpsRandom.java
import java.io.*;
import java.util.Random;

public class RpsRandom {
    static char[] MOVES = {'R', 'P', 'S'};
    // Solutions should be deterministic, so we need to set our random seed
    static Random rand = new Random(1337);
    
    static char playMove() {
        return MOVES[rand.nextInt(3)];
    }
    
    static void readBotMove(char botMove) {
        // This strategy doesn't care about opponent moves, so we do nothing here
    }
    
    static BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
    static PrintWriter out = new PrintWriter(System.out);

    public static void main(String[] args) throws IOException {
        int T = Integer.parseInt(in.readLine());
        for (int i = 0; i < T; i++) {
            out.println(playMove());
            out.flush();
            readBotMove(in.readLine().charAt(0));
        }
    }
}
